daemon/src/model.rs

```diff
@@ -603,7 +603,7 @@ impl Timestamp {
 }
 
 /// Funding rate per SETTLEMENT_INTERVAL
-#[derive(Debug, Clone, Copy, PartialEq)]
+#[derive(Debug, Clone, Copy, PartialEq, Serialize, Deserialize)]
```

Comment:
Simply deriving these will allow you to side-step the invariants you are checking in the constructor. Deriving `Serialize` / `Deserialize` for a new-type is almost always incorrect :)

Reply:
please bear in mind that the invariants are temporary, until we support charging the maker. Happy to hear how would I go about it if I was planning to have long-term invariants in a type! Custom serialise/deserialise with fallbacking to an option if the invariants are not upheld?

Reply:
> fallbacking to an option if the invariants are not upheld?

No, you can just fail the deserialization with an arbitrary message! Usually, you'd just call the constructor to make sure you don't duplicate them and then just do `map_err(D::Error::custom)?;` to forward the error.